parlai/scripts/train_model.py

```diff
@@ -264,46 +302,23 @@ def run_eval(valid_worlds, opt, datatype, max_exs=-1, write_log=False):
     print('[ running eval: ' + datatype + ' ]')
     reports = []
     for v_world in valid_worlds:
-        v_world.reset()
-        cnt = 0
-        while not v_world.epoch_done():
-            v_world.parley()
-            if cnt == 0 and opt['display_examples']:
-                print(v_world.display() + '\n~~')
-                print(v_world.report())
-            cnt += v_world.opt['batchsize']
-            if max_exs > 0 and cnt >= max_exs / len(valid_worlds):
-                # note this max_exs is approximate--some batches won't always be
-                # full depending on the structure of the data
-                break
-        valid_report = v_world.report()
-        reports.append(valid_report)
-    v_world.reset()  # this makes sure agent doesn't remember valid data
-
-    if len(valid_worlds) > 1:
-        # multiple tasks, aggregate metrics
-        tasks = [world.opt['task'] for world in valid_worlds]
-        agent_metrics = {}
-        total_report = {'tasks': {}}
-        for i, report in enumerate(reports):
-            total_report['tasks'][tasks[i]] = report
-            for metric, val in report.items():
-                agent_metrics.setdefault(metric, []).append(val)
-        total_report['tasks']['all'] = {}
-        for metric, vals in agent_metrics.items():
-            total_report['tasks']['all'][metric] = \
-                round_sigfigs(sum(vals) / len(vals), 4)
-        valid_report = total_report
-
-    metrics = datatype + ':' + str(valid_report)
+        task_report = _run_single_eval(opt, v_world,
+                                       max_exs / len(valid_worlds))
```

Comment:
does this really need to wrap?

Jason already blacked this file; you should black it too. `black -l 88 -S filename.py`